### PR TITLE
S3 support

### PR DESCRIPTION
**Pull request by haiwei-li**

### CODE REVIEW

**Comment by haiwei-li** [comment]
@lifeix please help to review, thanks!

File: devicemodel/hw/platform/acpi/acpi.c [hunk 1]
```diff
@@ -93,6 +93,12 @@
 #define RTCT_OFFSET		0xF00
 #define DSDT_OFFSET		0x1100
 
+/* Define the byte offset and byte length in FACS table */
+#define WAKING_VECTOR_OFFSET   12
+#define WAKING_VECTOR_LEN      4
+#define X_WAKING_VECTOR_OFFSET 24
+#define X_WAKING_VECTOR_LEN    8
+
```
Comment (lifeix) [inline devicemodel/hw/platform/acpi/acpi.c:101]: Are X_WAKING_VECTOR_XXX needed ?

Comment (haiwei-li) [reply devicemodel/hw/platform/acpi/acpi.c:101]: No, I will remove

**Review by lifeix — COMMENTED** [review]
This patch should be the first one.

**Comment by yuhuanX** [comment]
start to run premerge test

**Review by lifeix — APPROVED** [review]

**Review by chejianj — APPROVED** [review]

**Comment by yuhuanX** [comment]
start to run premerge test

**Comment by yuhuanX** [comment]
start to run premerge test

**Comment by yuhuanX** [comment]
start to run premerge test

**Comment by yuhuanX** [comment]
start to run premerge test

**Comment by yuhuanX** [comment]
start to run premerge test

**Comment by yuhuanX** [comment]
start to run premerge test

**Comment by yuhuanX** [comment]
start to run premerge test

**Comment by yuhuanX** [comment]
start to run premerge test

**Comment by yuhuanX** [comment]
start to run premerge test

**Comment by yuhuanX** [comment]
start to run premerge test

**Pull request merged**